### DISCUSSION TITLE
How to get the model shema and all fields settings?

### DISCUSSION THREAD

**Discussion by marcusdiy** [prisma/prisma#14777] [Q&A] [answered]
Hi, I need to get all the field configuration to use in an custom crud module wich relies on fields. 
It's a pain to rewrite the fields and their type, size, etc...
I've been relying on _baseDmmf but it seems to be discouraged as it's an internal value that may change any moment.
`let schema = prisma._baseDmmf.datamodel.models.filter(x => x.name === 'Users')[0];
`
but it gives incomplete info, like primaryKey is always null and there is no table name...
Is there some method to get this data? Something like `await prisma.getSchema()` would be indeed cool.

Let me know. Thanks

**Comment by marcusdiy** [comment]
Just found about **valentinpalkovic/prisma-json-schema-generator** but it's not an official package and seems to generate the schema just right but it has incolmpete data, like there is no indication on what should be the primary key and as always table name and other parameter usefull for crud propouses

**Comment by marcusdiy** [comment]
I need to access the schema to filter out the arguments. Its a shame that there is no way to access the fields info. Seems oddly strange.

**Comment by austincrim** [comment]
Hey @marcusdiy 👋,

If what you need is not available in the DMMF, I'm not sure what options we have.

I would love to hear more about your use case, maybe we can find an alternate solution.

**Reply by marcusdiy** [reply]
I need to have access to primary key and constraits as i have an CRUD system wich relays on the primary id value. Currenlty I use this:
```
function getPrismaSchema(model) {
    let schema_model_name = model[0].toUpperCase() + model.slice(1);
    let schema = prisma._baseDmmf.datamodel.models.filter(x => x.name === schema_model_name)[0];
    schema = Utils.extend({}, schema); // remove references
    let fields = {};
    for (var fidx in schema.fields) fields[schema.fields[fidx].name] = schema.fields[fidx];
    schema.fields = fields;
    return schema;
}
```
So if I do `getPrismaSchema('account')` the result is:
```
{
  "name": "Account",
  "dbName": null,
  "fields": {},
  "primaryKey": null,
  "uniqueFields": [],
  "uniqueIndexes": [],
  "isGenerated": false
}
```

**Comment by marcusdiy** [comment]
The main use is in a CRUD generator, so i can just provide the model name.
The script will fetch all the fields of the model and will create an editable form, so I dont have to define them all manually.
This way the form will automatically update even if I change the schema later on.

**Comment by austincrim** [comment, marked as answer]
I think your best bet is to open a new  [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with all the relevant info and our engineering teams will consider it from there 👍.

**Reply by marcusdiy** [reply]
https://github.com/prisma/prisma/issues/14943

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
